specification/alertsmanagement/resource-manager/Microsoft.AlertsManagement/preview/2023-05-01-privatepreview/AlertProcessingRules.json

```diff
@@ -0,0 +1,865 @@
+        "systemData": {
+          "description": "Alert processing rule system data.",
+          "$ref": "../../../../../common-types/resource-management/v2/types.json#/definitions/systemData"
+        }
+      }
+    },
+    "AlertProcessingRuleProperties": {
+      "type": "object",
+      "description": "Alert processing rule properties defining scopes, conditions and scheduling logic for alert processing rule.",
+      "required": [
+        "scopes",
+        "actions"
+      ],
+      "properties": {
+        "scopes": {
+          "description": "Scopes on which alert processing rule will apply.",
+          "$ref": "#/definitions/Scopes"
+        },
+        "conditions": {
+          "description": "Conditions on which alerts will be filtered.",
+          "$ref": "#/definitions/Conditions"
+        },
+        "schedule": {
+          "description": "Scheduling for alert processing rule.",
+          "$ref": "#/definitions/Schedule"
+        },
+        "actions": {
+          "type": "array",
+          "description": "Actions to be applied.",
+          "items": {
+            "$ref": "#/definitions/Action"
+          }
+        },
+        "description": {
+          "type": "string",
+          "description": "Description of alert processing rule."
+        },
+        "enabled": {
+          "type": "boolean",
+          "default": true,
+          "description": "Indicates if the given alert processing rule is enabled or disabled."
+        }
+      }
+    },
+    "Action": {
+      "type": "object",
+      "description": "Action to be applied.",
+      "discriminator": "actionType",
+      "required": [
+        "actionType"
+      ],
+      "properties": {
+        "actionType": {
+          "type": "string",
+          "description": "Action that should be applied.",
+          "enum": [
+            "AddActionGroups",
+            "RemoveAllActionGroups",
+            "CorrelateAlerts"
+          ],
+          "x-ms-enum": {
+            "name": "actionType",
+            "modelAsString": true
+          }
+        }
+      }
+    },
+    "AddActionGroups": {
+      "x-ms-discriminator-value": "AddActionGroups",
+      "type": "object",
+      "description": "Add action groups to alert processing rule.",
+      "required": [
+        "actionGroupIds"
+      ],
+      "allOf": [
+        {
+          "$ref": "#/definitions/Action"
+        }
+      ],
+      "properties": {
+        "actionGroupIds": {
+          "description": "List of action group Ids to add to alert processing rule.",
+          "type": "array",
+          "items": {
+            "type": "string"
+          }
+        }
+      }
+    },
+    "RemoveAllActionGroups": {
+      "x-ms-discriminator-value": "RemoveAllActionGroups",
+      "type": "object",
+      "description": "Indicates if all action groups should be removed.",
+      "allOf": [
+        {
+          "$ref": "#/definitions/Action"
+        }
+      ]
+    },
+    "CorrelateAlerts": {
```

Comment:
>CorrelateAlerts [](http://example.com/codeflow?start=5&length=15)

Should the name be more like: CorrelationMethods?

Reply:
thank you, our PM has decided on "CorrelateBy" after a discussion with the UI expert.